plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php

```diff
@@ -117,6 +123,8 @@ public function __destruct()
 
     public static function loadMappings()
     {
+        global $globalDiacritics;
```

Comment:
Could you just use the protected `$globalDiacritics` property declared earlier?

Reply:
since it's a static function, i can't use `$this` so i'm not sure how else to access `$globalDiacritics`, suggestions?

Reply:
It seems like something like this could work:

1. Move "Check for diacritics_mapping.yml" code into a new method (called `loadCharMapping` or `loadCharFilter` whatever makes sense)
2. Have this method return the results of `sfYaml::load`
3. In the `initialize` method set `$this->config['index']['configuration']['analysis']['char_filter']['diacritics_lowercase']` to the results of a call to this method